src/Drupal/Driver/Fields/Drupal8/ImageHandler.php

```diff
@@ -17,9 +17,7 @@ public function expand($values) {
     }
 
     /** @var \Drupal\file\FileInterface $file */
-    $file = file_save_data(
-      $data,
-      'public://' . uniqid() . '.jpg');
+    $file = \Drupal::service('file.repository')->writeData($data, 'public://' . uniqid() . '.jpg');
```

Comment:
This has been introduced in Drupal 9.3, so this breaks compatibility with Drupal 8 and Drupal 9 up to version 9.2.x

I am not opposed to this, but if this gets in we have to clearly stop supporting Drupal 9.2 and below. We need to add this to `composer.json`:

```
"conflict": {
    "drupal/core": ">=8.0 <9.3"
},
```

This will ensure that any projects stuck on an old version of Drupal for any reason will not accidentally break if they update DrupalDriver. They will remain on the last supported version.

Reply:
added